### ISSUE TITLE
Label not found for custom dataset

### ISSUE THREAD

**Issue by swaingotnochill** [ultralytics/yolov5#2736]
## ❔Question
I am following the same exact method for organizing the data for custom tutorial. I want to test using the pretrained weights. But when I run test.py, it cannot  find the labels, hence I cannot find the mAP.
![Screenshot from 2021-04-08 20-31-11](https://user-images.githubusercontent.com/54025913/114050943-500fb100-98aa-11eb-85a5-3535f64120f3.png)

**Comment by glenn-jocher** [comment]
@swaingotnochill yes that's right, test.py is not finding your val labels in your data.yaml. Please make sure that your directories and data.yaml are organized as in the Train Custom Data tutorial:
https://docs.ultralytics.com/yolov5/tutorials/train_custom_data

## Train On Custom Data

### 1. Create dataset.yaml

[COCO128](https://www.kaggle.com/datasets/ultralytics/coco128) is a small tutorial dataset composed of the first 128 images in [COCO](http://cocodataset.org/#home) train2017. These same 128 images are used for both training and validation to verify our training pipeline is capable of overfitting. [data/coco128.yaml](https://github.com/ultralytics/yolov5/blob/master/data/coco128.yaml), shown below, is the dataset configuration file that defines 1) an optional download command/URL for auto-downloading, 2) a path to a directory of training images (or path to a *.txt file with a list of training images), 3) the same for our validation images, 4) the number of classes, 5) a list of class names:
```yaml
# download command/URL (optional)
download: https://github.com/ultralytics/yolov5/releases/download/v1.0/coco128.zip

# train and val data as 1) directory: path/images/, 2) file: path/images.txt, or 3) list: [path1/images/, path2/images/]
train: ../coco128/images/train2017/
val: ../coco128/images/train2017/

# number of classes
nc: 80

# class names
names: ['person', 'bicycle', 'car', 'motorcycle', 'airplane', 'bus', 'train', 'truck', 'boat', 'traffic light',
        'fire hydrant', 'stop sign', 'parking meter', 'bench', 'bird', 'cat', 'dog', 'horse', 'sheep', 'cow',
        'elephant', 'bear', 'zebra', 'giraffe', 'backpack', 'umbrella', 'handbag', 'tie', 'suitcase', 'frisbee',
        'skis', 'snowboard', 'sports ball', 'kite', 'baseball bat', 'baseball glove', 'skateboard', 'surfboard',
        'tennis racket', 'bottle', 'wine glass', 'cup', 'fork', 'knife', 'spoon', 'bowl', 'banana', 'apple',
        'sandwich', 'orange', 'broccoli', 'carrot', 'hot dog', 'pizza', 'donut', 'cake', 'chair', 'couch',
        'potted plant', 'bed', 'dining table', 'toilet', 'tv', 'laptop', 'mouse', 'remote', 'keyboard', 
        'cell phone', 'microwave', 'oven', 'toaster', 'sink', 'refrigerator', 'book', 'clock', 'vase', 'scissors', 
        'teddy bear', 'hair drier', 'toothbrush']
```


### 3. Organize Directories

Organize your train and val images and labels according to the example below. In this example we assume `/coco128` is **next to** the `/yolov5` directory. **YOLOv5 locates labels automatically for each image** by replacing the last instance of `/images/` in each image path with `/labels/`. For example: 
```bash
dataset/images/im0.jpg  # image
dataset/labels/im0.txt  # label
```

<p align="center"><img width="698" src="https://user-images.githubusercontent.com/26833433/112467887-e18a0980-8d67-11eb-93af-6505620ff8aa.png"></p>

**Comment by swaingotnochill** [comment]
In data.yaml , I mentioned the path like this
val: ../cityperson/images


The thing is it is reading images, but not the labels..I followed the same style as well...Still, the error persists
@glenn-jocher

**Comment by glenn-jocher** [comment]
@swaingotnochill if your images are in `../cityperson/images` then your labels should be in `../cityperson/labels`

**Comment by glenn-jocher** [comment]
@swaingotnochill if you look at your screenshot it says scanning citypersons/labels, but you should not be passing the labels directory in your yaml, only the images directories. The labels dir is automatically determined from your images dir.

**Comment by swaingotnochill** [comment]
@glenn-jocher naah..I haven't passed labels directory...If you look on right side of screenshot, you can see the data.yaml file....besides, I also tried your approach previously and made necessary changes in data.yaml file, still it couldn't find the labels...

**Comment by glenn-jocher** [comment]
@swaingotnochill hmm, well I'm not sure what could be the problem then. Note the labels must have the same filename as the images except with a *.txt extension. 

I would start from the COCO128 example and work from there, making small changes or simply replacing the COCO128 data with your own.
